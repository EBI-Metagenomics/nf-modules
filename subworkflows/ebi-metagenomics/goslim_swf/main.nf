
include { GENERATEGAF                              } from '../../../modules/ebi-metagenomics/generategaf/main'
include { OWLTOOLS                                 } from '../../../modules/ebi-metagenomics/owltools/main'
include { MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS  } from '../../../modules/ebi-metagenomics/mgnifypipelinestoolkit/summarisegoslims/main'

workflow GOSLIM_SWF {

    take:
    ch_ips     // channel: [ val(meta), path(tsv) ]
    go_obo     // channel: path(obo)
    goslim_ids // channel: path(txt)
    go_banding // channel: path(txt)

    main:

    ch_versions = Channel.empty()

    GENERATEGAF( ch_ips )
    ch_versions = ch_versions.mix(GENERATEGAF.out.versions.first())

    OWLTOOLS(
        GENERATEGAF.out.gaf,
        go_obo,
        goslim_ids
    )
    ch_versions = ch_versions.mix(OWLTOOLS.out.versions.first())

    MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS(
        ch_ips,
        OWLTOOLS.out.gaf,
        go_obo,
        go_banding
    )
    ch_versions = ch_versions.mix(MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS.out.versions.first())

    emit:
    go_summary            = MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS.out.go_summary     // channel: [ val(meta), path(csv) ]
    goslim_summary        = MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS.out.goslim_summary // channel: [ val(meta), path(csv) ]
    versions              = ch_versions                                                // channel: [ versions.yml ]
}

