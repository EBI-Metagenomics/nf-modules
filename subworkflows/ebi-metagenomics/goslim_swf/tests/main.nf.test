
nextflow_workflow {

    name "Test Subworkflow GOSLIM_SWF"
    script "../main.nf"
    workflow "GOSLIM_SWF"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_ebimetagenomics"
    tag "subworkflows/goslim_swf"
    tag "generategaf"
    tag "owltools"
    tag "mgnifypipelinestoolkit/summarisegoslims"

    test("goslim_swf") {

        when {
            workflow {
                """
                input[0] = channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/generategaf/tests/ips_out.tsv", checkIfExists: true)
                ])
                input[1] = file("${baseDir}/modules/ebi-metagenomics/owltools/tests/go-dummy.obo", checkIfExists: true)
                input[2] = file("${baseDir}/modules/ebi-metagenomics/owltools/tests/goslim_ids.txt", checkIfExists: true)
                input[3] = file("${baseDir}/modules/ebi-metagenomics/mgnifypipelinestoolkit/summarisegoslims/tests/goslim_banding_2024.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
