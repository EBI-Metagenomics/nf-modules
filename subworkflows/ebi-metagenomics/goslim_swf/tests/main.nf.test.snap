{
    "goslim_swf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.tsv:md5,c24ea2f0b0763052a18190b77a61bd42"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.tsv:md5,c2ce302a04dd3cfa52488edf0b08744a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,9803d78712ef1091ff5f785e34209e66",
                    "versions.yml:md5,ba96e7433636f8b76ab87a3c196129da",
                    "versions.yml:md5,f7674bdfeee9a4d98dbb33e474dd7a68"
                ],
                "go_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.tsv:md5,c24ea2f0b0763052a18190b77a61bd42"
                    ]
                ],
                "goslim_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.tsv:md5,c2ce302a04dd3cfa52488edf0b08744a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9803d78712ef1091ff5f785e34209e66",
                    "versions.yml:md5,ba96e7433636f8b76ab87a3c196129da",
                    "versions.yml:md5,f7674bdfeee9a4d98dbb33e474dd7a68"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-27T22:28:53.816878"
    }
}